File: src/main/java/com/cuupa/classificator/services/kb/semantic/dataExtraction/IbanExtract.java
package com.cuupa.classificator.services.kb.semantic.dataExtraction;

import org.jetbrains.annotations.NotNull;

import java.util.regex.Pattern;

public class IbanExtract extends Extract {

	private static Pattern pattern;

	public IbanExtract(@NotNull String regex) {
		pattern = Pattern.compile(regex);
	}

	@Override
	public Pattern getPattern() {
		return pattern;
	}

	@NotNull
	@Override
	public String normalize(String value) {
		value = value.replace(" ", "");
		StringBuilder sb = new StringBuilder();
		char[] charArray = value.toCharArray();
		for (int i = 0; i < charArray.length; i++) {
			if (i % 4 == 0 && i > 0) {
				sb.append(" ");
			}
			sb.append(charArray[i]);
		}
		return sb.toString();
	}
}

File: src/main/java/com/cuupa/classificator/services/kb/semantic/dataExtraction/IbanExtract.kt
package com.cuupa.classificator.services.kb.semantic.dataExtraction

import com.cuupa.classificator.constants.StringConstants
import org.apache.logging.log4j.util.Strings
import java.util.regex.Pattern

class IbanExtract(regex: String) : Extract(Pattern.compile(regex.trim(), Pattern.CASE_INSENSITIVE)) {

    override fun normalize(value: String): String {
        val charArray = value.replace(StringConstants.blank, Strings.EMPTY).toCharArray()
        val sb = StringBuilder()
        for (i in charArray.indices) {
            if (i > 0 && i % 4 == 0) {
                sb.append(Strings.EMPTY)
            }
            sb.append(charArray[i])
        }
        return sb.toString()
    }

    companion object {
        const val name = "[IBAN]"
    }
}